src/bun.js/bindings/bindings.cpp

```diff
@@ -229,6 +229,45 @@ AsymmetricMatcherResult matchAsymmetricMatcher(JSGlobalObject* globalObject, JSC
             }
         }
 
+        return AsymmetricMatcherResult::FAIL;
+    } else if (auto* expectArrayContaining = jsDynamicCast<JSExpectArrayContaining*>(matcherPropCell)) {
+        JSValue expectedArrayValue = expectArrayContaining->m_arrayValue.get();
+
+        if (otherProp.isCell() && otherProp.asCell()->type() == ArrayType) {
+            if (expectedArrayValue.isCell() && expectedArrayValue.asCell()->type() == ArrayType) {
+                JSArray* expectedArray = jsDynamicCast<JSArray*>(expectedArrayValue);
+                JSArray* otherArray = jsDynamicCast<JSArray*>(otherProp);
+
+                unsigned expectedLength = expectedArray->length();
+                unsigned otherLength = otherArray->length();
+
+                // A empty array is all array's subset
+                if (expectedLength == 0) {
+                    return AsymmetricMatcherResult::PASS;
+                }
+
+                // O(m*n) but works for now
+                for (unsigned m = 0; m < expectedLength; m++) {
+                    JSValue expectedValue = expectedArray->get(globalObject, m);
+                    bool found = false;
+
+                    for (unsigned n = 0; n < otherLength; n++) {
+                        JSValue otherValue = otherArray->get(globalObject, n);
+                        if (JSValue::strictEqual(globalObject, expectedValue, otherValue)) {
```

Comment:
Are you sure it's `strictEqual` and not loose/deep equal?

Reply:
I check jest code, that should use deep equal, fix it already.

Reply:
The other thing: arrays can have holes in them. This needs to check if `otherValue.isEmpty()` (it is represented as `0`, which is separate from `undefined` or `null`)